docs/rollup-boost-ha.md

```diff
@@ -0,0 +1,328 @@
+
+Below is a happy path sequence diagram illustrating how `rollup-boost` facilitates payload construction/validation:
+
+```mermaid
+sequenceDiagram
+    participant CL as op-node
+    participant RB as rollup-boost
+    participant EL as op-geth
+    participant B as Builder
+
+    %% FCU forwarded with payload attrs
+    Note over CL: FCU with Attributes
+    CL->>RB: engine_forkChoiceUpdated(..., Attrs)
+    RB->>EL: engine_forkChoiceUpdated(..., Attrs)
+    RB->>B: engine_forkChoiceUpdated(..., Attrs)
+    B->>B: Start payload job
+
+    %% get_payload from builders
+    CL->>RB: engine_getPayload(PayloadId)
+    RB->>EL: engine_getPayload(PayloadId)
+    RB->>B: engine_getPayload(PayloadId)
+    B-->>RB: {executionPayload, blockValue}
+    EL-->>RB: {executionPayload, blockValue}
+
+    %% block selection policy
+    RB->>RB: Optimistically select builder block 
+
+    %% Validate best payload
+    Note over RB, EL: Validate builder block
+    RB->>EL: engine_newPayload(ExecutionPayload)
+    EL->>RB: {status: VALID, ...}
+
+    %% Propagate block to builders
+    Note over RB: Propagate new block to builder
+    RB->>B: engine_newPayload(ExecutionPayload)
+
+    %% Return final payload to op-node
+    RB->>CL: {executionPayload, blockValue}
+
+```
+
+### Failure Scenarios
+
+<details>
+<summary> Failure Scenarios</summary>
+
+Below is a high level summary of how each failure scenario is handled. All existing failure modes assumed by upstream `op-conductor` are maintained:
+
+| Failure Scenario | Category | Scenario and Solution |
+| --- | --- | --- |
+| Leader Sequencer Fails | Sequencer Failure | Conductor will detect sequencer failure and start to transfer leadership to another node, which will start sequencing instead. Currently,`op-conductor` will not know if the newly elected sequencer has a healthy builder. In the event that `op-conductor` selects a sequencer with an unhealthy builder, manual intervention is needed to either switch the leader or recover the builder. Once the originally failed sequencer is back online, it will join the sequencer set as a follower. |
+| Leader `rollup-boost` Fails | Rollup Boost Failure | Leader sequencer `rollup-boost` fails, causing `op-conductor`s sequencer health checks to fail, notifying conductor to elect a new leader. This failure mode is the same as a typical leader sequencer failure. Once the sequencer `rollup-boost` recovers, the sequencer will rejoin the sequencer set as a follower. |
```

Comment:
to be clear, the sequencer is always in the cluster as follower, it's just that it won't start sequencing since it's not considered healthy

Reply:
Thanks for flagging, this has been updated.